spec/compiler/semantic/new_spec.cr

```diff
@@ -224,4 +224,26 @@ describe "Semantic: new" do
       ),
       "wrong number of arguments for 'Bar.new' (given 0, expected 1)"
   end
+
+  it "uses correct receiver for `initialize` in namespaced classes (#4086)" do
+    assert_type %(
+      class Foo
+        class Baz(T)
```

Comment:
The generic argument shouldn't be necessary here, right?

Reply:
It is necessary here because if the owner type is not generic then `new`'s body would simply use `allocate` (with implicit self) instead of `Baz.allocate`.